### ISSUE TITLE
How to reference a project in another namespace

### ISSUE THREAD

**Issue by cebidhem** [mongodb/mongodb-atlas-kubernetes#215]
Hi there, 

To give a bit of context, we have 2 use cases for Atlas: 
- multi-tenant clusters
- single tenant clusters

Usually for both cases, we may or may not put several clusters in the same projects. 

Therefore, we may decide to have global objects (`atlasprojects.atlas.mongodb.com`, or in multi-tenant clusters cases also `atlasclusters.atlas.mongodb.com`) in a common namespace which would be `atlas-system`, and the dedicated resources to specific applications in the application namespaces `app1-ns` or `app2-ns` (for example, `atlasdatabaseusers.atlas.mongodb.com` or dedicated `atlasclusters.atlas.mongodb.com`).

Now when trying to apply a `atlasprojects.atlas.mongodb.com` or `atlasclusters.atlas.mongodb.com` in a different namespace than the project, the operator can't find the reference:
```
2021-04-21T08:54:46.809Z	INFO	controllers.AtlasDatabaseUser	Status update	{"atlasdatabaseuser": "app1-ns/my-db-user", "lastCondition": {"type":"DatabaseUserReady","status":"False","lastTransitionTime":null,"reason":"InternalError","message":"AtlasProject.atlas.mongodb.com \"my-atlas-project\" not found"}}
```
This log mean the spec.projectRef must be in the same namespace than the object we're trying to create.
I tried to use `app1-ns/my-atlas-project` and `my-atlas-project.app1-ns` but it didn't make the trick :)

Do you think it would be possible to add a namespace reference for a `atlasprojects.atlas.mongodb.com` object ? As in
```
spec:
  projectRef:
    name: my-atlas-project
    namespace: app1-ns
```

The thing is, we could create all the resources inside the `atlas-system` namespace, but since secrets are namespaced resources then we couldn't use the secretRef as described here https://docs.atlas.mongodb.com/reference/atlas-operator/atlasdatabaseuser-custom-resource/#connection-secrets

Thanks!

**Comment by theburi** [comment]
This makes sense. We will add it to the backlog.

**Comment by jasonmimick** [comment]
We have added this request to the active backlog and will deliver in an upcoming release. Thanks for your patience.

**Comment by igor-karpukhin** [comment]
@cebidhem  This is already implemented in 1.0.0. See https://github.com/mongodb/mongodb-atlas-kubernetes/blob/main/pkg/api/v1/atlasdeployment_types.go#L50. Here is an example of referencing a project by deployment: 
```
  projectRef:
    name: my-project
    namespace: mongodb-atlas-system
```
